### PR TITLE
Adding Complementary Map feature

### LINKED ISSUE #31: Feature: Complementary channel
Given a map (quantum channel) compute the complementary map. 

As an example, this functionality is present in the QETLAB package. While the implementation there is in MATLAB, the spirit of the implementation should be quite similar to the implementation for `toqito`:
http://www.qetlab.com/ComplementaryMap

Write the functionality for this task in `toqito/channel_ops/complimentary_channel.py` and ensure proper unit tests for this feature are written in `tests/test_channel_ops/test_complimentary_channel.py`. Also ensure that the automated docs are updated by placing the line `toqito.channel_ops.complimentary_channel` in `docs/channels.rst` under `Operations on Quantum Channels`.

### PR DESCRIPTION
**Pull request by Shivansh20128**
## Description
Provide a brief description of the PR's purpose here. If your PR is supposed to fix an existing issue, use
a [keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to link your PR to the issue.
This PR adds a new feature in the channel operations module to compute the complementary channel/map as discussed in the issue mentioned.
#Resolves #31

## Changes
Notable changes that this PR has either accomplished or will accomplish. Feel free to add more lines to the itemized list
below.

  -  [x] Added function `complementary_channel`.
  -  [x] Added docstrings for the function.
  -  [x] Added tests for the function `complementary_channel`.

## Checklist
Before marking your PR ready for review, make sure you checked the following locally. If this is your first PR, you might be notified of some workflow failures after a maintainer has approved the workflow jobs to be run on your PR. 

Additional information is available in the [documentation](https://toqito.readthedocs.io/en/latest/contributing.html#testing).

  -  [x] Use `ruff` for errors related to code style and formatting.
  -  [x] Verify all previous and newly added unit tests pass in `pytest`.
  -  [x] Check the documentation build does not lead to any failures. `Sphinx` build can be checked locally for any failures related to your PR
  -  [x] Use `linkcheck` to check for broken links in the documentation
  -  [x] Use `doctest` to verify the examples in the function docstrings work as expected.

### CODE REVIEW

**Comment by Shivansh20128** [comment]
Hi @vprusso, can you look at the function for `complementary_channels` and the test cases I added and review them?
Thank You

**Review by vprusso — APPROVED** [review]
LGTM! I'll ping @purva-thakre to see if she is able to give this a look as well. Great work, @Shivansh20128 , and my apologies for taking a bit longer than usual to get to this. All the best!

File: toqito/channel_ops/complementary_channel.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    (Section: Representations and Characterizations of Channels from :cite:`Watrous_2018_TQI`).
+
+    The complementary map is derived from the given quantum channel's Kraus operators by
+    rearranging the rows of the input Kraus operators into the Kraus operators of the
+    complementary map.
+
+    Specifically, for each Kraus operator :math:`K_i` in the input channel :math:`\Phi`,
+    we define the complementary Kraus operators :math:`K_i^C` by stacking the rows of
+    :math:`K_i` from all Kraus operators vertically.
+
```
Comment (purva-thakre) [inline toqito/channel_ops/complementary_channel.py:18]: this docstring is missing an example

Comment (Shivansh20128) [reply toqito/channel_ops/complementary_channel.py:18]: Ahh yes, I am adding the docstring here as well.

File: toqito/channel_ops/complementary_channel.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    """
+    num_kraus = len(kraus_ops)
+    if num_kraus==0:
+        raise ValueError("All Kraus operators must be non-empty matrices.")
+
+    op_dim = kraus_ops[0].shape[0]
+
+    if any(k.shape[0] != k.shape[1] for k in kraus_ops):
+        raise ValueError("All Kraus operators must be square matrices.")
```
Comment (purva-thakre) [inline toqito/channel_ops/complementary_channel.py:37]: kraus operators must satisfy other properties in addition to being square matrices. 
These should be used to check the user has not provided invalid kraus operators.

Comment (Shivansh20128) [reply toqito/channel_ops/complementary_channel.py:84]: Ohh, can you name these properties?
Thank you

Comment (purva-thakre) [reply toqito/channel_ops/complementary_channel.py:84]: See Chapter 8 of Nielsen and Chuang.

Comment (Shivansh20128) [reply toqito/channel_ops/complementary_channel.py:84]: @purva-thakre Please help me with this.
Thank you

Comment (Shivansh20128) [reply toqito/channel_ops/complementary_channel.py:84]: Hi! I have added the above checks for the Kraus operators.

**Comment by vprusso** [comment]
This LGTM, but I'll ping @purva-thakre as well for this. Thank you, @Shivansh20128 !

**Review by vprusso — APPROVED** [review]

**Comment by vprusso** [comment]
Thanks again for the contribution, @Shivansh20128 ! Apologies for my delay in merging!

**Pull request merged**